app/controllers/api/v1/materials_controller.rb

```diff
@@ -307,8 +347,288 @@ def assign_to_class
+                                                                        id )
+    if existing.nil?
+        render json: {  :message => "No existing standard statement to delete." },
+                        :status => 200
+    end
+
+    existing.destroy
+
+    render json: {  :message => "Successfully removed standard." },
+                    :status => 200
+  end
+
+
+  #
+  # Get standard statements from ASN query
+  #
+  # params[:asn_document_id]                 The document URI
+  # params[:asn_statement_notation_query]    The statement notation to match
+  # params[:asn_statement_label_query]       The statement label to match
+  # params[:asn_description_query]           Text in the description to match
+  #
+  # Optionally supply material_type and material_id to populate
+  # the is_applied property.
+  #
+  # params[:start]                          The start index
+  #
+  def get_standard_statements
+
+    key                     = ENV['ASN_API_KEY']
+    document_id             = params[:asn_document_id]
+    statement_notation_q    = params[:asn_statement_notation_query]
+    statement_label_q       = params[:asn_statement_label_query]
+    description_q           = params[:asn_description_query]
+
+    material_type           = params[:material_type]
+    material_id             = params[:material_id]
+
+    start                   = params[:start]
+
+    applied_map = {}
+
+    #
+    # See if we can populate is_applied data
+    #
+    if material_type.present? && material_id.present?
+        statements = StandardStatement.find_all_by_material_type_and_material_id(material_type, material_id)
+
+        statements.map { |s| applied_map[s.uri] = true }
+    end
+
+    # 
+    # Build query string
+    #
+    query_string = "(and is_part_of:'#{document_id}' type:'Statement'"
+
+    #
+    # Notation doesn't seem to match wildcards (this is a "literal" field
+    # type, though I thought the documentation says it allows wildcards in 
+    # literal fields...)
+    #
+    if statement_notation_q.present?
+        query_string << " statement_notation:'#{statement_notation_q}'"
+    end
+
+    if statement_label_q.present?
+        query_string << " statement_label:'#{statement_label_q}'"
+    end
+
+    if description_q.present?
+        query_string << " description:'*#{description_q}*'"
+    end
+
+    query_string << ")"
+
+    #
+    # Sort order.
+    # Notes:
+    # list_id is null on NGSS.
+    #
+    rank = "identifier"
+
+    #
+    # See http://toolkit.asn.desire2learn.com/documentation/asn-search
+    #
+    query = {   "bq"            => query_string,
+                "return-fields" => @@asn_return_fields,
+                "rank"          => rank,
+                "key"           => "#{key}" }
+
+    if start.present?
+        query["start"] = start
+    end
+
+    response = HTTParty.get(@@ASN_SEARCH_BASE_URL, :query => query)
+
+    puts "*** Reponse #{response}"
+
+    count = response["hits"]["found"]
```

Comment:
What happens here if the search fails due to a server error and there is no `response["hits"]` entry?